Bug fix for bad return value when getting a node from the catalog

taharah opened the pull request.

taharah commented:
Test results...
```
Run options: include {:focus=>true}

All examples were filtered out; ignoring {:focus=>true}

Randomized with seed 28009
Diplomat::Node .......
Diplomat::Datacenter ...
Diplomat::Maintenance .......
Diplomat::Health .........
Diplomat::Lock ......
Diplomat .......
Diplomat::Event .................
Diplomat::Kv .........................
Diplomat::Check .......
Diplomat::Members .
Diplomat::Service ....................
Diplomat::Acl .......
Diplomat::Session ....
Diplomat::RestClient ........
Diplomat::Nodes ..

Finished in 0.09017 seconds (files took 0.27352 seconds to load)
130 examples, 0 failures

Randomized with seed 28009
```

PeterFCaswell commented:
Thanks for making this change, @taharah, but I'm worried that it is potentially a breaking change. Can you bump the version number, please, and add a test that the value returned is indeed an OpenStruct?

taharah commented:
@PeterFCaswell I have bumped the version and updated the tests per your request. I only did a minor version since I don't believe this is a breaking change (as shown below). I initiated this as a bug fix since the documentation states that the return value should be an OpenStruct, and thus, why I was surprised to be getting a hash during some of my tests.

Updated tests:
```
Run options: include {:focus=>true}

All examples were filtered out; ignoring {:focus=>true}

Randomized with seed 60365
Diplomat::Service ....................
Diplomat::Acl .......
Diplomat::Kv .........................
Diplomat::Session ....
Diplomat::Nodes ..
Diplomat::Health .........
Diplomat::Lock ......
Diplomat::Datacenter ...
Diplomat::Check .......
Diplomat::Node ........
Diplomat::Members .
Diplomat .......
Diplomat::RestClient ........
Diplomat::Event .................
Diplomat::Maintenance .......

Finished in 0.10872 seconds (files took 0.26484 seconds to load)
131 examples, 0 failures

Randomized with seed 60365
```

Breaking change test:
```
[30] pry(main)> test = Diplomat::Node.get('foobar')
=> {"Node"=>{"Node"=>"foobar", "Address"=>"10.10.10.10", "TaggedAddresses"=>{"wan"=>"10.10.10.10"}, "CreateIndex"=>393321, "ModifyIndex"=>2068267},
 "Services"=>
  {"TESTAGAIN_solr1"=>
    {"ID"=>"TESTAGAIN_solr1", "Service"=>"TESTAGAIN", "Tags"=>["master", "TESTAGAIN", "solr1", "Virtual", "dev"], "Address"=>"", "Port"=>2003, "EnableTagOverride"=>false, "CreateIndex"=>2011668, "ModifyIndex"=>2028009},
   "TESTAGAIN_solr2"=>
    {"ID"=>"TESTAGAIN_solr2", "Service"=>"TESTAGAIN", "Tags"=>["master", "TESTAGAIN", "solr2", "Virtual", "dev"], "Address"=>"", "Port"=>2004, "EnableTagOverride"=>false, "CreateIndex"=>2011661, "ModifyIndex"=>2028052},
   "TESTAGAIN_solr3"=>
    {"ID"=>"TESTAGAIN_solr3", "Service"=>"TESTAGAIN", "Tags"=>["master", "TESTAGAIN", "solr3", "Virtual", "dev"], "Address"=>"", "Port"=>2005, "EnableTagOverride"=>false, "CreateIndex"=>2011662, "ModifyIndex"=>2028061},
   "TESTME-solr1"=>{"ID"=>"TESTME-solr1", "Service"=>"TESTME-solr1", "Tags"=>["slave", "TESTME", "solr1", "Virtual", "dev"], "Address"=>"", "Port"=>2006, "EnableTagOverride"=>false, "CreateIndex"=>1824910, "ModifyIndex"=>2028056},
   "TESTME-solr2"=>{"ID"=>"TESTME-solr2", "Service"=>"TESTME-solr2", "Tags"=>["slave", "TESTME", "solr2", "Virtual", "dev"], "Address"=>"", "Port"=>2007, "EnableTagOverride"=>false, "CreateIndex"=>1824944, "ModifyIndex"=>2028033},
   "TESTME-solr3"=>{"ID"=>"TESTME-solr3", "Service"=>"TESTME-solr3", "Tags"=>["slave", "TESTME", "solr3", "Virtual", "dev"], "Address"=>"", "Port"=>2008, "EnableTagOverride"=>false, "CreateIndex"=>1824978, "ModifyIndex"=>2028063},
   "TESTUS-solr1"=>{"ID"=>"TESTUS-solr1", "Service"=>"TESTUS-solr1", "Tags"=>["master", "TESTUS", "solr1", "Virtual", "dev"], "Address"=>"", "Port"=>2001, "EnableTagOverride"=>false, "CreateIndex"=>1064743, "ModifyIndex"=>2068254},
   "TESTUS-solr2"=>{"ID"=>"TESTUS-solr2", "Service"=>"TESTUS-solr2", "Tags"=>["slave", "TESTUS", "solr2", "Virtual", "dev"], "Address"=>"", "Port"=>2002, "EnableTagOverride"=>false, "CreateIndex"=>1064784, "ModifyIndex"=>2068267}}}
[31] pry(main)> struct = OpenStruct.new(test)
=> #<OpenStruct Node={"Node"=>"foobar", "Address"=>"10.10.10.10", "TaggedAddresses"=>{"wan"=>"10.10.10.10"}, "CreateIndex"=>393321, "ModifyIndex"=>2068267}, Services={"TESTAGAIN_solr1"=>{"ID"=>"TESTAGAIN_solr1", "Service"=>"TESTAGAIN", "Tags"=>["master", "TESTAGAIN", "solr1", "Virtual", "dev"], "Address"=>"", "Port"=>2003, "EnableTagOverride"=>false, "CreateIndex"=>2011668, "ModifyIndex"=>2028009}, "TESTAGAIN_solr2"=>{"ID"=>"TESTAGAIN_solr2", "Service"=>"TESTAGAIN", "Tags"=>["master", "TESTAGAIN", "solr2", "Virtual", "dev"], "Address"=>"", "Port"=>2004, "EnableTagOverride"=>false, "CreateIndex"=>2011661, "ModifyIndex"=>2028052}, "TESTAGAIN_solr3"=>{"ID"=>"TESTAGAIN_solr3", "Service"=>"TESTAGAIN", "Tags"=>["master", "TESTAGAIN", "solr3", "Virtual", "dev"], "Address"=>"", "Port"=>2005, "EnableTagOverride"=>false, "CreateIndex"=>2011662, "ModifyIndex"=>2028061}, "TESTME-solr1"=>{"ID"=>"TESTME-solr1", "Service"=>"TESTME-solr1", "Tags"=>["slave", "TESTME", "solr1", "Virtual", "dev"], "Address"=>"", "Port"=>2006, "EnableTagOverride"=>false, "CreateIndex"=>1824910, "ModifyIndex"=>2028056}, "TESTME-solr2"=>{"ID"=>"TESTME-solr2", "Service"=>"TESTME-solr2", "Tags"=>["slave", "TESTME", "solr2", "Virtual", "dev"], "Address"=>"", "Port"=>2007, "EnableTagOverride"=>false, "CreateIndex"=>1824944, "ModifyIndex"=>2028033}, "TESTME-solr3"=>{"ID"=>"TESTME-solr3", "Service"=>"TESTME-solr3", "Tags"=>["slave", "TESTME", "solr3", "Virtual", "dev"], "Address"=>"", "Port"=>2008, "EnableTagOverride"=>false, "CreateIndex"=>1824978, "ModifyIndex"=>2028063}, "TESTUS-solr1"=>{"ID"=>"TESTUS-solr1", "Service"=>"TESTUS-solr1", "Tags"=>["master", "TESTUS", "solr1", "Virtual", "dev"], "Address"=>"", "Port"=>2001, "EnableTagOverride"=>false, "CreateIndex"=>1064743, "ModifyIndex"=>2068254}, "TESTUS-solr2"=>{"ID"=>"TESTUS-solr2", "Service"=>"TESTUS-solr2", "Tags"=>["slave", "TESTUS", "solr2", "Virtual", "dev"], "Address"=>"", "Port"=>2002, "EnableTagOverride"=>false, "CreateIndex"=>1064784, "ModifyIndex"=>2068267}}>
[32] pry(main)> struct['Node']
=> {"Node"=>"foobar", "Address"=>"10.10.10.10", "TaggedAddresses"=>{"wan"=>"10.10.10.10"}, "CreateIndex"=>393321, "ModifyIndex"=>2068267}
```
